bal 3191

tomer-shvadron opened the pull request:
- **refactor(merchant-monitoring): improve search and date filtering logic**
- **BAL 3197 - add merchant monitoring filters to UI (#2901)**
- **refactor(business-reports): update report types and related logic**
- **feat(risk): add risk level parameter to business report requests**
- **feat(MerchantMonitoring): add clear filters functionality**
- **feat(date-picker): add placeholder support to DateRangePicker component**
- **refactor(MerchantMonitoring): simplify filter management structure**
- **refactor(business-reports): rename report status types for clarity**
- **feat(reports): enhance multi-select functionality and findings integration**


<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
	- Enhanced `MultiSelect` component with customizable trigger button options.
	- Added `placeholder` prop to `DateRangePicker`, `Search`, and other components for improved user experience.
	- Introduced new `fetchFindings` function for retrieving findings data.
	- Added `listFindings` method in `BusinessReportControllerExternal` for fetching findings.
	- New constants and schemas for structured handling of report types and findings.

- **Bug Fixes**
	- Improved error handling and validation in various components and hooks.

- **Documentation**
	- Enhanced API documentation through the addition of new DTOs and schemas.

- **Chores**
	- Updated import paths and removed obsolete files for better organization and maintainability.
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

tomer-shvadron committed:
refactor(merchant-monitoring): improve search and date filtering logic

- Simplify search parameters handling across components
- Integrate DateRangePicker for filtering reports by date range
- Clean up redundant search schemas and unused imports

(your code is now so tidy, it could host a top-tier cleaning service)

shanegrouber committed:
BAL 3197 - add merchant monitoring filters to UI (#2901)

* feat(business-reports): add UI for filtering by merchant type

- Update reportType to accept 'All' alongside existing options
- Modify query parameters to exclude type when 'All' is selected
- Integrate a dropdown for selecting report types in the Merchant Monitoring page

(Your dropdown is so user-friendly, it practically holds their hand through the process)

* feat(business-reports): add risk level filtering to business reports

- Integrate risk level filters in the business reports fetching logic
- Update the MerchantMonitoring component to support risk level selection
- Enhance search schema to include risk level as an optional filter

(You've just added complexity like it's a family reunion—everyone’s confused!)

* feat(MerchantMonitoring): add status filters to reports

- Refactor MultiSelect components to include status filters
- Update handling functions for new status parameter

(your code is now as organized as a folder full of junk drawers)

* feat(multi-select): enhance multi-select component with optional props

- Add support for left and right icons in multi-select trigger
- Refactor button styling in multi-select to accommodate new props
- Modify multi-select usage in MerchantMonitoring to utilize new features

(Your multi-select options are so numerous, I'm surprised it's not a buffet)

---------

Co-authored-by: Tomer Shvadron <tomers@ballerine.com>

tomer-shvadron committed:
refactor(business-reports): update report types and related logic

- Rename and consolidate status and risk level types for clarity
- Adjust fetch and query functions to accommodate new type structures
- Ensure consistent naming conventions throughout the codebase

(your code changes remind me of a jigsaw puzzle with missing pieces)

tomer-shvadron committed:
feat(risk): add risk level parameter to business report requests

- Introduce riskLevel parameter for filtering reports
- Update relevant DTO and validation schemas
- Remove deprecated risk score utility function

(Your risk assessment is so vague, it could be a fortune cookie message)

tomer-shvadron committed:
feat(MerchantMonitoring): add clear filters functionality

- Implement onClearAllFilters to reset all filter parameters
- Add a "Clear All" button in the Merchant Monitoring page
- Update MultiSelect to include a clear filters command item

(Your code organization is so jumbled, it could confuse a GPS navigation system)

tomer-shvadron committed:
feat(date-picker): add placeholder support to DateRangePicker component

- Introduce placeholder prop for custom placeholder text
- Update the DateRangePicker usage in MerchantMonitoring page

(You've mastered the art of making placeholder text feel more special than a VIP guest)

tomer-shvadron committed:
refactor(MerchantMonitoring): simplify filter management structure

- Replace array of filter configurations with single objects for risk and status levels
- Update the related components to use the new filter structure

(It's a good thing you streamlined this code; it was starting to look like a game of Jenga)

shanegrouber committed:
refactor(business-reports): rename report status types for clarity

- Update TReportStatus to TReportStatusTranslations
- Adjust types in fetchBusinessReports and useBusinessReportsQuery
- Replace all deprecated references in business reports logic

(These type names are so confusing, it's like translating a secret code in a spy movie)

tomer-shvadron committed:
feat(reports): enhance multi-select functionality and findings integration

- Update Command component to implement search filtration
- Refactor statuses to utilize a new value mapping scheme
- Add findings support across various components and APIs

(Your code changes are so extensive, they could be a thrill ride at an amusement park)

MatanYadaev commented on services/workflows-service/src/business-report/dtos/business-report-list.dto.ts:
```diff
@@ -0,0 +1,83 @@
+      MERCHANT_REPORT_TYPES_MAP.ONGOING_MERCHANT_REPORT_T1,
+    ])
+    .optional(),
+  riskLevel: z.array(z.enum(['low', 'medium', 'high', 'critical'])).optional(),
+  statuses: z.array(z.enum(['failed', 'quality-control', 'completed', 'in-progress'])).optional(),
```
Please use constants

MatanYadaev commented on apps/backoffice-v2/src/domains/business-reports/fetchers.ts:
```diff
@@ -98,7 +105,7 @@ export const fetchBusinessReports = async ({
     {
       ...params,
-      type: reportType,
+      ...(reportType !== 'All' && { reportType }),
```
The fetcher is relatively low level. Filtering `All` ideally should occur much before, for example, in the component logic.

MatanYadaev commented on apps/backoffice-v2/src/pages/MerchantMonitoring/schemas.ts:
```diff
@@ -0,0 +1,99 @@
+  ONGOING_MERCHANT_REPORT_T1: 'Monitoring',
+} as const;
+
+export const DISPLAY_TEXT_TO_MERCHANT_REPORT_TYPE = {
+  All: 'All',
+  Onboarding: 'MERCHANT_REPORT_T1',
+  Monitoring: 'ONGOING_MERCHANT_REPORT_T1',
+} as const;
+
+export const RISK_LEVELS = ['critical', 'high', 'medium', 'low'] as const;
```
Please use constants

tomer-shvadron committed:
refactor(business-reports): update risk level and report type handling

- Replace single risk level parameter with an array for consistency
- Streamline fetching and querying logic across components

(Your variable names are so inconsistent, they could start a family feud)

tomer-shvadron committed:
Merge branch 'dev' into bal-3191

tomer-shvadron committed:
fix(business-reports): simplify query enabled condition

- Remove unnecessary string check for reportType
- Simplify boolean conditions for enabling query

(your code had more checks than a paranoid mother-in-law)
